src/gwc/src/main/java/org/geoserver/gwc/GWCSynchEnv.java

```diff
@@ -0,0 +1,85 @@
+/* (c) 2022 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.gwc;
+
+import java.util.Properties;
+import org.geoserver.platform.GeoServerEnvironment;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geowebcache.GeoWebCacheEnvironment;
+import org.springframework.beans.BeansException;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/** This class is responsible for synchronizing the GeoServer environment for GWC. */
+public class GWCSynchEnv implements ApplicationContextAware {
+    private GeoWebCacheEnvironment gwcEnvironment;
+
+    GeoServerEnvironment gsEnvironment = GeoServerExtensions.bean(GeoServerEnvironment.class);
```

Comment:
Why not inject this one with the constructor?

Reply:
Fixed